esphome/components/spi/spi.h

```
@@ -80,6 +80,11 @@ class SPIComponent : public Component {
 
   template<SPIBitOrder BIT_ORDER, SPIClockPolarity CLOCK_POLARITY, SPIClockPhase CLOCK_PHASE>
   void read_array(uint8_t *data, size_t length) {
+    if (this->miso_ == nullptr) {
```

If miso is not defined, I think we should print an error or something, not silently ignore it.

---

I will add a ESP_LOGE

---

You're right - we can't log here without creating a helper method that logs it in spi.cpp

Ok, I guess not logging is ok - unfortunately the logging has to be removed here. having the global `TAG` symbol in the header file includes the symbol in every file it's used in (increased code size and other issues)